app/src/main/java/org/kiwix/kiwixmobile/language/viewmodel/SaveLanguagesAndFinish.kt

```diff
@@ -33,7 +33,7 @@ data class SaveLanguagesAndFinish(
     Flowable.fromCallable { languageDao.insert(languages) }
       .subscribeOn(Schedulers.io())
       .subscribe({
-        activity.finish()
+        activity.onBackPressed()
```

Comment:
Unsure about this. What's best? Popping the backstack or artificially calling onBackPressed?

Reply:
OnBackPressed is fine I think